src/admin/shared/components/ContentPicker/ContentPicker.tsx

```diff
@@ -73,86 +68,76 @@ const ContentPicker: FunctionComponent<ContentPickerProps> = ({
 					);
 				});
 		}
-	}, [currentTypes, t]);
+	}, [currentType, t]);
 
 	const onChangeText = (value: string) => {
 		setInput(value);
 		onSelect(parsePickerItem('EXTERNAL_LINK', value));
 	};
 
-	const onChangeType = (currentValues: ValueType<PickerTypeOption>) => {
-		setCurrentTypes((currentValues as PickerTypeOption[]) || []);
+	const onChangeType = (selected: ValueType<PickerTypeOption>) => {
+		setCurrentType(selected as PickerTypeOption);
 	};
 
 	const renderGroupLabel = (data: any) => <span>{data.label}</span>;
 
-	const renderSelectPicker = () => (
-		<Grid>
-			<Column size="1">
-				<ReactSelect
-					{...REACT_SELECT_DEFAULT_OPTIONS}
-					id="content-picker-type"
-					placeholder={t('admin/content/components/content-picker/content-picker___type')}
-					options={typeOptions}
-					isSearchable={false}
-					isMulti={true}
-					isOptionDisabled={(option: PickerTypeOption) => !!option.disabled}
-					onChange={onChangeType}
-				/>
-			</Column>
-			<Column size="3">
-				<ReactSelect
-					{...REACT_SELECT_DEFAULT_OPTIONS}
-					id="content-picker-query"
-					placeholder={t('admin/content/components/content-picker/content-picker___item')}
-					formatGroupLabel={renderGroupLabel}
-					options={groupedOptions as any}
-					isSearchable={false}
-					isDisabled={!currentTypes.length}
-					isLoading={loading}
-					onChange={(selectedItem: ValueType<PickerItem>) => {
-						onSelect(get(selectedItem, 'value'));
-					}}
-				/>
-			</Column>
-		</Grid>
-	);
-
-	const renderInputPicker = () => <TextInput value={input} onChange={onChangeText} />;
+	const renderContentPickerControls = () => {
+		if (!currentType) {
+			return null;
+		}
 
-	const renderEditor = () => {
-		switch (controls) {
-			case 'content':
+		switch (currentType.picker) {
+			case 'SELECT':
 				return renderSelectPicker();
-			case 'external-url':
-				return renderInputPicker();
+			case 'TEXT_INPUT':
+				return renderTextInputPicker();
 			default:
 				return null;
 		}
 	};
 
+	const renderSelectPicker = () => (
+		<ReactSelect
+			{...REACT_SELECT_DEFAULT_OPTIONS}
+			id="content-picker-query"
+			placeholder={t('admin/content/components/content-picker/content-picker___item')}
+			formatGroupLabel={renderGroupLabel}
+			options={options as any}
+			isSearchable={false}
+			isLoading={loading}
+			onChange={(selectedItem: ValueType<PickerItem>) => {
+				onSelect(get(selectedItem, 'value'));
+			}}
+			value={
+				options.find(
+					option => get(option, 'value.value', '') === get(currentSelection, 'value', '')
```

Comment:
wat gebeurt er als beide options niet gedefinieerd zijn? dan returned hij true
is dat de bedoeling?

Reply:
In principe zal dit niet voorvallen, maar ik heb er toch een kleine fix voor geïmplementeerd.